service/history/transferQueueStandbyProcessor.go

```diff
@@ -177,10 +185,25 @@ func (t *transferQueueStandbyProcessorImpl) processActivityTask(transferTask *pe
 			return nil
 		}
 
+		pushToMatching := time.Now().Sub(transferTask.GetVisibilityTimestamp()) > t.shard.GetConfig().StandbyClusterDelay()
```

Comment:
We should use clock of the remote cluster to compute 5 minute delay in deciding when to move task to matching engine.